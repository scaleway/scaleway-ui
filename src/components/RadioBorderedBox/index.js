import styled from '@emotion/styled'
import { transparentize } from 'polished'
import PropTypes from 'prop-types'
import React from 'react'
import Badge from '../Badge'
import BorderedBox from '../BorderedBox'
import Box from '../Box'
import Radio from '../Radio'

const StyledBorderedBox = styled(BorderedBox)`
  &:hover {
    cursor: pointer;
  }

  ${({ disabled, checked, theme: { colors } }) => {
    if (disabled)
      return `
        cursor: not-allowed !important;
        color: ${colors.gray300};
      `
    if (checked)
      return `
        border: 1px solid ${colors.primary} !important;
        box-shadow: 0 0 0 2px ${transparentize(0.75, colors.primary)};
      `

    return null
  }}
`

const RadioBorderedBox = ({
  label,
  labelDescription,
  badgeText,
  badgeSize,
  badgeVariant,
  checked,
  onChange,
  onFocus,
  onBlur,
  disabled,
  name,
  value,
  size,
  children,
  ...props
}) => (
  <StyledBorderedBox
    as="label"
    display="block"
    disabled={disabled}
    checked={checked}
    {...props}
  >
    <Box display="flex" alignItems="center" mb={1}>
      <Radio
        name={name}
        checked={checked}
        onChange={onChange}
        onFocus={onFocus}
        onBlur={onBlur}
        disabled={disabled}
        value={value}
        size={size}
        mr="4px"
      >
        {label}
      </Radio>
      <Box as="span" mr={1}>
        {labelDescription}
      </Box>
      {badgeText && (
        <Badge size={badgeSize} variant={badgeVariant}>
          {badgeText}
        </Badge>
      )}
    </Box>
    <Box>{children}</Box>
  </StyledBorderedBox>
)

RadioBorderedBox.propTypes = {
  label: PropTypes.string.isRequired,
  labelDescription: PropTypes.string,
  badgeText: PropTypes.string,
  badgeVariant: PropTypes.string,
  badgeSize: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  size: PropTypes.number,
}

RadioBorderedBox.defaultProps = {
  labelDescription: undefined,
  size: 24,
  disabled: false,
  checked: false,
  onFocus: null,
  onBlur: null,
  badgeText: undefined,
  badgeVariant: 'info',
  badgeSize: 'xsmall',
}

export default RadioBorderedBox
