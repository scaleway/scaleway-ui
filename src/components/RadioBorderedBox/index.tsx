import styled from '@emotion/styled'
import { transparentize } from 'polished'
import PropTypes from 'prop-types'
import React, { FunctionComponent, InputHTMLAttributes, ReactNode } from 'react'
import Badge, { badgeSizes, badgeVariants } from '../Badge'
import type { Sizes, Variants } from '../Badge'
import Box, { XStyledProps } from '../Box'
import Radio from '../Radio'

const StyledBox = styled(Box)<{ disabled: boolean; checked: boolean }>`
  ${({ disabled, checked, theme: { colors } }) => {
    if (disabled)
      return `
        cursor: not-allowed !important;
        color: ${colors.gray300};
      `
    if (checked)
      return `
        border: 1px solid ${colors.primary} !important;
        box-shadow: 0 0 0 2px ${transparentize(0.75, colors.primary)};
      `

    return null
  }}
`

type RadioBorderedBoxProps = {
  badgeSize?: Sizes
  badgeText?: string
  badgeVariant?: Variants
  children: ReactNode
  label: ReactNode
  labelDescription?: string
  name: string
  size?: number
  value: string | number
} & InputHTMLAttributes<HTMLInputElement> &
  XStyledProps

const RadioBorderedBox: FunctionComponent<RadioBorderedBoxProps> = ({
  label,
  labelDescription,
  badgeText,
  badgeSize = 'xsmall',
  badgeVariant = 'info',
  checked = false,
  onChange,
  onFocus,
  onBlur,
  disabled = false,
  name,
  value,
  size = 24,
  children,
  ...props
}) => (
  <StyledBox
    bordered
    display="block"
    disabled={disabled}
    checked={checked}
    {...props}
  >
    <Box display="flex" alignItems="center" mb={1}>
      <Radio
        name={name}
        checked={checked}
        onChange={onChange}
        onFocus={onFocus}
        onBlur={onBlur}
        disabled={disabled}
        value={value}
        size={size}
        mr="4px"
      >
        {label}
      </Radio>
      <Box as="span" mr={1}>
        {labelDescription}
      </Box>
      {badgeText && (
        <Badge size={badgeSize} variant={badgeVariant}>
          {badgeText}
        </Badge>
      )}
    </Box>
    <div>{children}</div>
  </StyledBox>
)

RadioBorderedBox.propTypes = {
  /**
   * See API of `Badge` component
   */
  badgeSize: PropTypes.oneOf<Sizes>(badgeSizes),

  /**
   * Add a badge next to the label
   */
  badgeText: PropTypes.string,
  /**
   * See API of `Badge` component
   */
  badgeVariant: PropTypes.oneOf<Variants>(badgeVariants),
  checked: PropTypes.bool,
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  /**
   * Label next to the radio button, can be a string or a more complex child
   */
  label: PropTypes.node.isRequired,
  /**
   * Description next to the label
   */
  labelDescription: PropTypes.string,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  /**
   * Size of the radio button
   */
  size: PropTypes.number,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
}

export default RadioBorderedBox
