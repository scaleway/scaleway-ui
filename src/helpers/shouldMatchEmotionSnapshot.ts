/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
