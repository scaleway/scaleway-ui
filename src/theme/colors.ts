const white = '#fff'
const gray50 = '#fafafb'
const gray100 = '#f6f5f7'
const gray200 = '#eeeeff'
const gray300 = '#dce1eb'
const gray350 = '#d4dae7'
const gray550 = '#b2b6c3'
const gray700 = '#4a4f62'
const gray950 = '#151a2d'
const black = '#000'

const ngray300 = '#dbdbdf'

const violet = '#4f0599'
const lightViolet = '#a365f6'

const green = '#45d6b5'
const red = '#ef5774'
const orange = '#ff8c69'
const blue = '#3f6ed8'
const darkBlue = '#2f52a1'

const zumthor = '#e4edff'
const foam = '#cdfcf1'
const serenade = '#ffefe6'
const pippin = '#ffe1e7'
const shadow = '#a5a5cd'
const gold = '#ffd536'
const crimson = '#cf0f34'

// System colors
const primary = violet
const beta = orange
const success = green
const warning = red
const info = blue

const transparent = 'transparent'

const colors = {
  beta,
  black,
  blue,
  crimson,
  darkBlue,
  foam,
  gold,
  gray50,
  gray100,
  gray200,
  gray300,
  gray350,
  gray550,
  gray700,
  gray950,
  green,
  info,
  lightViolet,
  ngray300,
  orange,
  pippin,
  primary,
  red,
  serenade,
  shadow,
  success,
  transparent,
  violet,
  warning,
  white,
  zumthor,
} as const

export type Color = keyof typeof colors

export default colors
