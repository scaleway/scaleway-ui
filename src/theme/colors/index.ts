import * as localContrasts from './contrasts'

export type ContrastType = {
  100: string
  200: string
  300: string
  400: string
  500: string
  600: string
  700: string
  800: string
  900: string
  1000?: string
}

type GenerateTokensProps = {
  sentiment: string
  contrast: ContrastType
  neutralContrast: ContrastType
}

// This function will generate all colors tokens using shade of contrasts
const generateTokens = ({
  sentiment,
  contrast,
  neutralContrast,
}: GenerateTokensProps) => {
  // Neutral is a particular color it has more shade and they are used differently than usual colors.
  if (sentiment === 'neutral') {
    return {
      /* eslint-disable sort-keys */

      // Background
      background: contrast[500],
      backgroundHover: contrast[600],
      backgroundDisabled: contrast[400],
      backgroundWeak: contrast[100],
      backgroundWeakHover: contrast[800],
      backgroundWeakDisabled: contrast[300],
      backgroundStrong: contrast[800],
      backgroundStrongHover: contrast[800],
      backgroundStrongDisabled: contrast[700],

      // Text
      text: contrast[800],
      textHover: contrast[900],
      textDisabled: contrast[600],
      textWeak: contrast[600],
      textWeakHover: contrast[600],
      textWeakDisabled: contrast[500],
      textStrong: contrast[100],
      textStrongHover: contrast[100],
      textStrongDisabled: contrast[100],

      // Border
      border: contrast[300],
      borderHover: contrast[300],
      borderDisabled: contrast[300],
      borderWeak: contrast[500],
      borderWeakHover: contrast[500],
      borderWeakDisabled: contrast[200],
      borderStrong: contrast[100],
      borderStrongHover: contrast[100],
      borderStrongDisabled: contrast[100],

      /* eslint-enable sort-keys */
    }
  }

  return {
    /* eslint-disable sort-keys */

    // Background
    background: contrast[300],
    backgroundHover: contrast[300],
    backgroundDisabled: contrast[300],
    backgroundWeak: neutralContrast[100],
    backgroundWeakHover: contrast[800],
    backgroundWeakDisabled: neutralContrast[100],
    backgroundStrong: contrast[800],
    backgroundStrongHover: contrast[800],
    backgroundStrongDisabled: contrast[800],

    // Text
    text: contrast[800],
    textHover: contrast[900],
    textDisabled: contrast[400],
    textWeak: contrast[800],
    textWeakHover: contrast[900],
    textWeakDisabled: contrast[400],
    textStrong: neutralContrast[100],
    textStrongHover: neutralContrast[100],
    textStrongDisabled: neutralContrast[100],

    // Border
    border: contrast[800],
    borderHover: contrast[900],
    borderDisabled: contrast[200],
    borderWeak: contrast[800],
    borderWeakHover: contrast[800],
    borderWeakDisabled: contrast[400],
    borderStrong: neutralContrast[100],
    borderStrongHover: neutralContrast[100],
    borderStrongDisabled: neutralContrast[100],

    /* eslint-enable sort-keys */
  }
}

export type Color = keyof typeof localContrasts

// This function get in parameter a shade of contrasts and return a well formatted design tokens
export const colorsTokens = (
  contrasts: Partial<Record<Color, Partial<ContrastType>>>,
) => {
  // We first get contrasts passed as parameter if some are missing we use local contrasts and deep merge them
  const deepMergedContrasts = Object.keys(localContrasts).reduce(
    (acc, contrast) => ({
      ...acc,
      [contrast]: {
        ...localContrasts[contrast as Color],
        ...contrasts[contrast as Color],
      },
    }),
    {},
  ) as Record<Color, ContrastType>

  // Then with list of contrasts we can now create our design tokens using generateTokens function
  return Object.keys(deepMergedContrasts).reduce(
    (acc, contrast) => ({
      ...acc,
      [contrast]: generateTokens({
        contrast: deepMergedContrasts[contrast as Color],
        neutralContrast: deepMergedContrasts.neutral,
        sentiment: contrast,
      }),
    }),
    {},
  ) as Record<Color, ReturnType<typeof generateTokens>>
}

const colors = colorsTokens(localContrasts)

export default colors
