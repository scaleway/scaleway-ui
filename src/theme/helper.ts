import { Color, SCWUITheme } from '.'

export const getTextDisabled = (theme: SCWUITheme, variant: Color) =>
  variant
    ? theme.colors[variant as Exclude<Color, 'neutral'>]?.textWeakDisabled ??
      theme.colors.neutral.textDisabled
    : theme.colors.neutral.textDisabled

export const getBackgroundDisabled = (theme: SCWUITheme, variant: Color) =>
  variant
    ? theme.colors[variant as Exclude<Color, 'neutral'>]
        ?.backgroundWeakDisabled ?? theme.colors.neutral.backgroundDisabled
    : theme.colors.neutral.backgroundDisabled

export const getBorderDisabled = (theme: SCWUITheme, variant: Color) =>
  variant
    ? theme.colors[variant as Exclude<Color, 'neutral'>]?.borderWeakDisabled ??
      theme.colors.neutral.borderDisabled
    : theme.colors.neutral.borderDisabled
