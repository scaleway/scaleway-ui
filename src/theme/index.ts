/* eslint-disable sort-keys */
import colorsDeprecated from './colors'

const radii = {
  none: '0',
  default: '4px',
  large: '8px',
}

const space = {
  0: '0',
  0.25: '2px',
  0.5: '4px',
  0.75: '6px',
  1: '8px',
  2: '16px',
  2.25: '18px',
  3: '24px',
  4: '32px',
  5: '40px',
  6: '48px',
  7: '56px',
  8: '64px',
  9: '72px',
} as const
export type Spaces = keyof typeof space

const screens = {
  xsmall: 0,
  small: 576,
  medium: 768,
  large: 992,
  xlarge: 1200,
} as const
export type ScreenSize = keyof typeof screens

const fonts = {
  monospace: "'Lucida Console', Monaco, 'Courier New', Courier, monospace",
  sansSerif: 'Asap, System, sans-serif',
} as const

const theme = {
  colorsDeprecated,
  fonts,
  space,
  screens,
  radii,
}

type SCWUITheme = typeof theme & {
  linkComponent?: unknown
}

export default theme

export { colorsDeprecated, space, radii, fonts, screens, SCWUITheme }
