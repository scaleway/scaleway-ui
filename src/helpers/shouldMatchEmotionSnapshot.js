/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
