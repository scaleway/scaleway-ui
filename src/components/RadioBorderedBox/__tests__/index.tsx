import { ThemeProvider } from '@emotion/react'
import { render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import RadioBorderedBox from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'
import theme from '../../../theme'

describe('RadioBorderedBox', () => {
  test('renders correctly', () =>
    shouldMatchEmotionSnapshot(
      <RadioBorderedBox
        label="Choice"
        onChange={() => {}}
        name="radio"
        value="choice"
      >
        Choice description
      </RadioBorderedBox>,
    ))

  test('renders correctly when disabled', () =>
    shouldMatchEmotionSnapshot(
      <RadioBorderedBox
        label="Choice"
        onChange={() => {}}
        name="radio"
        value="choice"
        disabled
      >
        Choice description
      </RadioBorderedBox>,
    ))

  test('renders correctly when checked', () =>
    shouldMatchEmotionSnapshot(
      <RadioBorderedBox
        label="Choice"
        onChange={() => {}}
        name="radio"
        value="choice"
        checked
      >
        Choice description
      </RadioBorderedBox>,
    ))

  test('renders correctly with label desc and badge', () =>
    shouldMatchEmotionSnapshot(
      <RadioBorderedBox
        label="Choice"
        labelDescription="(something)"
        badgeText="Badge"
        badgeVariant="warning"
        badgeSize="small"
        onChange={() => {}}
        name="radio"
        value="choice"
        disabled
      >
        Choice description
      </RadioBorderedBox>,
    ))

  test('renders correctly and triggers change on borderedbox click', async () => {
    let choice
    render(
      <ThemeProvider theme={theme}>
        <RadioBorderedBox
          checked={choice === 'choice1'}
          id="radiotest"
          label="Choice 1"
          name="radioborderedbox"
          value="choice1"
          onChange={evt => {
            choice = evt.target.value
          }}
        >
          Choice 1 description
        </RadioBorderedBox>
      </ThemeProvider>,
    )
    const box = document.getElementById('radiotest')
    userEvent.click(box)
    await waitFor(() => expect(choice).toBe('choice1'))
  })
})
