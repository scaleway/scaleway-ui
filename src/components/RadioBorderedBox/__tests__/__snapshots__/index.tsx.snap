// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBorderedBox renders correctly 1`] = `
<DocumentFragment>
  .cache-1l5bs5y {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
}

.cache-1l5bs5y.cache-1l5bs5y {
  display: block;
}

.cache-1l5bs5y:hover {
  cursor: pointer;
}

.cache-h0rkxm.cache-h0rkxm {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.cache-dt36sk {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  height: 32px;
  cursor: pointer;
}

.cache-dt36sk.cache-dt36sk {
  margin-right: 4px;
}

.cache-dt36sk:hover .e1duuirp2,
.cache-dt36sk:focus .e1duuirp2 {
  background-color: rgba(220,225,235,0.25);
  border-radius: 50%;
}

.cache-dt36sk:hover .e1duuirp2>.e1duuirp3,
.cache-dt36sk:focus .e1duuirp2>.e1duuirp3 {
  fill: #4f0599;
}

.cache-kwm2j4 {
  min-width: 32px;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 8px;
}

.cache-ldqu23 {
  fill: #dce1eb;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.cache-ldqu23.cache-ldqu23 {
  vertical-align: middle;
}

.cache-cujed6 {
  position: absolute;
  opacity: 0.01;
}

.cache-1sg0ylu.cache-1sg0ylu {
  margin-right: 8px;
}

<label
    class="e13z6rra0 cache-1l5bs5y e7ah3bn0"
  >
    <div
      class="cache-h0rkxm e7ah3bn0"
    >
      <label
        class="e1duuirp1 cache-dt36sk e7ah3bn0"
        for="radio-choice"
      >
        <div
          class="e1duuirp2 cache-kwm2j4 e7ah3bn0"
        >
          <svg
            class="e1duuirp3 etwatq50 cache-ldqu23 e7ah3bn0"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
            />
          </svg>
        </div>
        Choice
        <input
          aria-checked="false"
          class="cache-cujed6 e1duuirp0"
          id="radio-choice"
          name="radio"
          tabindex="0"
          type="radio"
          value="choice"
        />
      </label>
      <span
        class="cache-1sg0ylu e7ah3bn0"
      />
    </div>
    <div>
      Choice description
    </div>
  </label>
</DocumentFragment>
`;

exports[`RadioBorderedBox renders correctly when checked 1`] = `
<DocumentFragment>
  .cache-my9iyn {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  border: 1px solid #4f0599!important;
  box-shadow: 0 0 0 2px rgba(79,5,153,0.25);
}

.cache-my9iyn.cache-my9iyn {
  display: block;
}

.cache-my9iyn:hover {
  cursor: pointer;
}

.cache-h0rkxm.cache-h0rkxm {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.cache-dt36sk {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  height: 32px;
  cursor: pointer;
}

.cache-dt36sk.cache-dt36sk {
  margin-right: 4px;
}

.cache-dt36sk:hover .e1duuirp2,
.cache-dt36sk:focus .e1duuirp2 {
  background-color: rgba(220,225,235,0.25);
  border-radius: 50%;
}

.cache-dt36sk:hover .e1duuirp2>.e1duuirp3,
.cache-dt36sk:focus .e1duuirp2>.e1duuirp3 {
  fill: #4f0599;
}

.cache-kwm2j4 {
  min-width: 32px;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 8px;
}

.cache-kptj9p {
  fill: #4f0599;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.cache-kptj9p.cache-kptj9p {
  vertical-align: middle;
}

.cache-cujed6 {
  position: absolute;
  opacity: 0.01;
}

.cache-1sg0ylu.cache-1sg0ylu {
  margin-right: 8px;
}

<label
    class="e13z6rra0 cache-my9iyn e7ah3bn0"
  >
    <div
      class="cache-h0rkxm e7ah3bn0"
    >
      <label
        class="e1duuirp1 cache-dt36sk e7ah3bn0"
        for="radio-choice"
      >
        <div
          class="e1duuirp2 cache-kwm2j4 e7ah3bn0"
        >
          <svg
            class="e1duuirp3 etwatq50 cache-kptj9p e7ah3bn0"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,7A5,5 0 0,0 7,12A5,5 0 0,0 12,17A5,5 0 0,0 17,12A5,5 0 0,0 12,7Z"
            />
          </svg>
        </div>
        Choice
        <input
          aria-checked="true"
          checked=""
          class="cache-cujed6 e1duuirp0"
          id="radio-choice"
          name="radio"
          tabindex="0"
          type="radio"
          value="choice"
        />
      </label>
      <span
        class="cache-1sg0ylu e7ah3bn0"
      />
    </div>
    <div>
      Choice description
    </div>
  </label>
</DocumentFragment>
`;

exports[`RadioBorderedBox renders correctly when disabled 1`] = `
<DocumentFragment>
  .cache-2w7h54 {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  cursor: not-allowed!important;
  color: #dce1eb;
}

.cache-2w7h54.cache-2w7h54 {
  display: block;
}

.cache-2w7h54:hover {
  cursor: pointer;
}

.cache-h0rkxm.cache-h0rkxm {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.cache-1382gvk {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  height: 32px;
  cursor: pointer;
  color: #dce1eb;
  cursor: not-allowed;
}

.cache-1382gvk.cache-1382gvk {
  margin-right: 4px;
}

.cache-kwm2j4 {
  min-width: 32px;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 8px;
}

.cache-ldqu23 {
  fill: #dce1eb;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.cache-ldqu23.cache-ldqu23 {
  vertical-align: middle;
}

.cache-cujed6 {
  position: absolute;
  opacity: 0.01;
}

.cache-1sg0ylu.cache-1sg0ylu {
  margin-right: 8px;
}

<label
    class="e13z6rra0 cache-2w7h54 e7ah3bn0"
    disabled=""
  >
    <div
      class="cache-h0rkxm e7ah3bn0"
    >
      <label
        class="e1duuirp1 cache-1382gvk e7ah3bn0"
        disabled=""
        for="radio-choice"
      >
        <div
          class="e1duuirp2 cache-kwm2j4 e7ah3bn0"
        >
          <svg
            class="e1duuirp3 etwatq50 cache-ldqu23 e7ah3bn0"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
            />
          </svg>
        </div>
        Choice
        <input
          aria-checked="false"
          aria-disabled="true"
          class="cache-cujed6 e1duuirp0"
          disabled=""
          id="radio-choice"
          name="radio"
          style="pointer-events: none;"
          type="radio"
          value="choice"
        />
      </label>
      <span
        class="cache-1sg0ylu e7ah3bn0"
      />
    </div>
    <div>
      Choice description
    </div>
  </label>
</DocumentFragment>
`;

exports[`RadioBorderedBox renders correctly with label desc and badge 1`] = `
<DocumentFragment>
  .cache-2w7h54 {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  cursor: not-allowed!important;
  color: #dce1eb;
}

.cache-2w7h54.cache-2w7h54 {
  display: block;
}

.cache-2w7h54:hover {
  cursor: pointer;
}

.cache-h0rkxm.cache-h0rkxm {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.cache-1382gvk {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  height: 32px;
  cursor: pointer;
  color: #dce1eb;
  cursor: not-allowed;
}

.cache-1382gvk.cache-1382gvk {
  margin-right: 4px;
}

.cache-kwm2j4 {
  min-width: 32px;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 8px;
}

.cache-ldqu23 {
  fill: #dce1eb;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.cache-ldqu23.cache-ldqu23 {
  vertical-align: middle;
}

.cache-cujed6 {
  position: absolute;
  opacity: 0.01;
}

.cache-1sg0ylu.cache-1sg0ylu {
  margin-right: 8px;
}

.cache-12dkvvj {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ef5774;
  color: #fff;
  font-size: 12px;
  line-height: 24px;
  height: 24px;
}

<label
    class="e13z6rra0 cache-2w7h54 e7ah3bn0"
    disabled=""
  >
    <div
      class="cache-h0rkxm e7ah3bn0"
    >
      <label
        class="e1duuirp1 cache-1382gvk e7ah3bn0"
        disabled=""
        for="radio-choice"
      >
        <div
          class="e1duuirp2 cache-kwm2j4 e7ah3bn0"
        >
          <svg
            class="e1duuirp3 etwatq50 cache-ldqu23 e7ah3bn0"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
            />
          </svg>
        </div>
        Choice
        <input
          aria-checked="false"
          aria-disabled="true"
          class="cache-cujed6 e1duuirp0"
          disabled=""
          id="radio-choice"
          name="radio"
          style="pointer-events: none;"
          type="radio"
          value="choice"
        />
      </label>
      <span
        class="cache-1sg0ylu e7ah3bn0"
      >
        (something)
      </span>
      <span
        class="egvl2gp0 cache-12dkvvj e7ah3bn0"
      >
        Badge
      </span>
    </div>
    <div>
      Choice description
    </div>
  </label>
</DocumentFragment>
`;
